### ISSUE TITLE
Spyder 6.0.3 is not allowing me to open a py file 

### ISSUE THREAD

**Issue by rubie43** [spyder-ide/spyder#23272]
### What steps will reproduce the problem?

Hello,
Not sure how to help reproduce the problem because I'm just selecting a py file from my file explorer and I get the following error. It's pop-up window 11 when Spyder opens after selecting the file.

To open (the filepath) as a project with Spyder, pelase use spyder - p "Data".

Please help.

I literally uninstalled and reinstalled the standalone installer because before, I had Spyder 6.0.2, thinking that was the issue. ¯\_(ツ)_/¯

## Versions

* Spyder version: 6.0.3  (standalone)
* Python version: 3.11.10 64-bit
* Qt version: 5.15.8
* PyQt5 version: 5.15.9
* Operating System: Windows-10-10.0.22631-SP0

### Dependencies

```
# Mandatory:
aiohttp >=3.9.3                  :  3.11.10 (OK)
asyncssh >=2.14.0,<3.0.0         :  2.18.0 (OK)
atomicwrites >=1.2.0             :  1.4.1 (OK)
chardet >=2.0.0                  :  5.2.0 (OK)
cloudpickle >=0.5.0              :  3.1.0 (OK)
cookiecutter >=1.6.0             :  2.6.0 (OK)
diff_match_patch >=20181111      :  20241021 (OK)
github >=2.3.0                   :  2.5.0 (OK)
importlib_metadata >=4.6.0       :  8.5.0 (OK)
intervaltree >=3.0.2             :  3.1.0 (OK)
IPython >=8.13.0,<9.0.0,!=8.17.1 :  8.30.0 (OK)
jedi >=0.17.2,<0.20.0            :  0.19.2 (OK)
jellyfish >=0.7                  :  1.1.2 (OK)
jsonschema >=3.2.0               :  4.23.0 (OK)
keyring >=17.0.0                 :  25.5.0 (OK)
nbconvert >=4.0                  :  7.16.4 (OK)
numpydoc >=0.6.0                 :  1.8.0 (OK)
parso >=0.7.0,<0.9.0             :  0.8.4 (OK)
pexpect >=4.4.0                  :  4.9.0 (OK)
pickleshare >=0.4                :  0.7.5 (OK)
psutil >=5.3                     :  6.1.0 (OK)
pygments >=2.0                   :  2.18.0 (OK)
pylint >=3.1,<4                  :  3.3.2 (OK)
pylint_venv >=3.0.2              :  3.0.4 (OK)
pyls_spyder >=0.4.0              :  0.4.0 (OK)
pylsp >=1.12.0,<1.13.0           :  1.12.0 (OK)
pylsp_black >=2.0.0,<3.0.0       :  2.0.0 (OK)
pyuca >=1.2                      :  1.2 (OK)
qdarkstyle >=3.2.0,<3.3.0        :  3.2.3 (OK)
qstylizer >=0.2.2                :  0.2.4 (OK)
qtawesome >=1.3.1,<1.4.0         :  1.3.1 (OK)
qtconsole >=5.6.1,<5.7.0         :  5.6.1 (OK)
qtpy >=2.4.0                     :  2.4.2 (OK)
rtree >=0.9.7                    :  1.3.0 (OK)
setuptools >=49.6.0              :  75.6.0 (OK)
sphinx >=0.6.6                   :  8.1.3 (OK)
spyder_kernels >=3.0.2,<3.1.0    :  3.0.2 (OK)
superqt >=0.6.2,<1.0.0           :  0.6.7 (OK)
textdistance >=4.2.0             :  4.6.2 (OK)
three_merge >=0.1.1              :  0.1.1 (OK)
watchdog >=0.10.3                :  6.0.0 (OK)
yarl >=1.9.4                     :  1.18.3 (OK)
zmq >=24.0.0                     :  26.2.0 (OK)

# Optional:
cython >=0.21                    :  3.0.11 (OK)
matplotlib >=3.0.0               :  3.9.3 (OK)
numpy >=1.7                      :  2.2.0 (OK)
pandas >=1.1.1                   :  2.2.3 (OK)
scipy >=0.17.0                   :  1.14.1 (OK)
sympy >=0.7.3                    :  1.13.3 (OK)
```

### Environment

<details><summary>Environment</summary>

```
Package                       Version
----------------------------- --------------
aiohappyeyeballs              2.4.3
aiohttp                       3.10.9
aiosignal                     1.3.1
alabaster                     1.0.0
arrow                         1.3.0
astroid                       3.3.5
asttokens                     2.4.1
asyncssh                      2.17.0
atomicwrites                  1.4.1
attrs                         24.2.0
autopep8                      2.0.4
babel                         2.16.0
beautifulsoup4                4.12.3
binaryornot                   0.4.4
black                         24.10.0
bleach                        6.1.0
certifi                       2024.8.30
cffi                          1.17.1
chardet                       5.2.0
charset-normalizer            3.4.0
click                         8.1.7
cloudpickle                   3.0.0
colorama                      0.4.6
comm                          0.2.2
cookiecutter                  2.6.0
cryptography                  43.0.1
debugpy                       1.8.7
decorator                     5.1.1
defusedxml                    0.7.1
Deprecated                    1.2.14
diff-match-patch              20230430
dill                          0.3.9
docstring-to-markdown         0.15
docutils                      0.21.2
et-xmlfile                    1.1.0
executing                     2.1.0
fastjsonschema                2.20.0
flake8                        7.1.1
frozenlist                    1.4.1
idna                          3.10
imagesize                     1.4.1
importlib_metadata            8.5.0
inflection                    0.5.1
intervaltree                  3.1.0
ipykernel                     6.29.5
ipython                       8.28.0
isort                         5.13.2
jaraco.classes                3.4.0
jaraco.context                6.0.1
jaraco.functools              4.1.0
jedi                          0.19.1
jellyfish                     1.1.0
Jinja2                        3.1.4
jsonschema                    4.23.0
jsonschema-specifications     2024.10.1
jupyter_client                8.6.3
jupyter_core                  5.7.2
jupyterlab_pygments           0.3.0
keyring                       25.4.1
markdown-it-py                3.0.0
MarkupSafe                    3.0.1
matplotlib-inline             0.1.7
mccabe                        0.7.0
mdurl                         0.1.2
mistune                       3.0.2
more-itertools                10.5.0
multidict                     6.1.0
mypy-extensions               1.0.0
nbclient                      0.10.0
nbconvert                     7.16.4
nbformat                      5.10.4
nest-asyncio                  1.6.0
numpy                         2.1.2
numpydoc                      1.8.0
openpyxl                      3.1.5
packaging                     24.1
pandas                        2.2.3
pandocfilters                 1.5.1
parso                         0.8.4
pathspec                      0.12.1
pexpect                       4.9.0
pickleshare                   0.7.5
pip                           24.3.1
platformdirs                  4.3.6
pluggy                        1.5.0
prompt_toolkit                3.0.48
propcache                     0.2.0
psutil                        6.0.0
ptyprocess                    0.7.0
pure_eval                     0.2.3
pycodestyle                   2.12.1
pycparser                     2.22
pydocstyle                    6.3.0
pyflakes                      3.2.0
PyGithub                      2.4.0
Pygments                      2.18.0
PyJWT                         2.9.0
pylint                        3.3.1
pylint-venv                   3.0.3
pyls-spyder                   0.4.0
PyNaCl                        1.5.0
PyQt5                         5.15.11
PyQt5-Qt5                     5.15.2
PyQt5_sip                     12.15.0
PyQtWebEngine                 5.15.7
PyQtWebEngine-Qt5             5.15.2
pyreadstat                    1.2.7
python-dateutil               2.9.0.post0
python-lsp-black              2.0.0
python-lsp-jsonrpc            1.1.2
python-lsp-server             1.12.0
python-slugify                8.0.4
pytoolconfig                  1.3.1
pytz                          2024.2
pyuca                         1.2
pywin32                       307
pywin32-ctypes                0.2.3
PyYAML                        6.0.2
pyzmq                         26.2.0
QDarkStyle                    3.2.3
qstylizer                     0.2.3
QtAwesome                     1.3.1
qtconsole                     5.6.0
QtPy                          2.4.1
referencing                   0.35.1
requests                      2.32.3
rich                          13.9.2
rope                          1.13.0
rpds-py                       0.20.0
Rtree                         1.3.0
setuptools                    75.1.0
six                           1.16.0
snowballstemmer               2.2.0
sortedcontainers              2.4.0
soupsieve                     2.6
Sphinx                        8.1.0
sphinxcontrib-applehelp       2.0.0
sphinxcontrib-devhelp         2.0.0
sphinxcontrib-htmlhelp        2.1.0
sphinxcontrib-jsmath          1.0.1
sphinxcontrib-qthelp          2.0.0
sphinxcontrib-serializinghtml 2.0.0
spyder-kernels                3.0.0
stack-data                    0.6.3
superqt                       0.6.7
tabulate                      0.9.0
text-unidecode                1.3
textdistance                  4.6.3
three-merge                   0.1.1
tinycss2                      1.3.0
tomli                         2.0.2
tomlkit                       0.13.2
tornado                       6.4.1
traitlets                     5.14.3
types-python-dateutil         2.9.0.20241003
typing_extensions             4.12.2
tzdata                        2024.2
ujson                         5.10.0
urllib3                       2.2.3
watchdog                      5.0.3
wcwidth                       0.2.13
webencodings                  0.5.1
whatthepatch                  1.0.6
wrapt                         1.16.0
yapf                          0.40.2
yarl                          1.14.0
zipp                          3.20.2

```
</details>

**Comment by ccordoba12** [comment]
Hey @rubie43, thanks for reporting. I think this problem happens because you're trying to open a file that has spaces in its name (e.g. `my file.py`) or in a directory with spaces (e.g. `C:\Users\My user\myfile.py`)

Could you confirm if that's the case?

**Comment by rubie43** [comment]
Hey @ccordoba12. Thank you for your help.

It looks like the filepath has spaces. I can replace the spaces with a '_' for the file name, but unfortunately, I cannot do so for the remainder of the filepath. Any suggestions?  Thank you.

**Comment by ccordoba12** [comment]
> It looks like the filepath has spaces. I can replace the spaces with a '_' for the file name, but unfortunately, I cannot do so for the remainder of the filepath

Ok, thanks for the confirmation.

> Any suggestions? Thank you.

The only possible thing you can do for now is to drag the file from the Windows file explorer and drop it in Spyder. Sorry, I know that's not ideal but there's no other workaround.

**Comment by ccordoba12** [comment]
I'm going to close this issue as a duplicate of #22844. We'll try to fix it in our next version (6.0.4), to be released in a few weeks.
